qa/README.rst

```diff
@@ -0,0 +1,46 @@
+health-ok.sh
+============
+
+ceph-bootstrap integration test automation script
+
+
+Overview
+--------
+
+This bash script contains integration tests for validating Ceph deployments
+done with ceph-bootstrap.
+
+The idea is to run this script with the appropriate arguments on the
+Salt Master node after using ceph-bootstrap to deploy a cluster.
+
+The script makes a number of assumptions, as listed under "Assumptions", below.
+
+On success (HEALTH_OK is reached, sanity tests pass), the script returns 0.
+On failure, for whatever reason, the script returns non-zero.
+
+The script produces verbose output on stdout, which can be captured for later
+forensic analysis.
+
+Though referred to as a bash script, ``health-ok.sh`` is only the entry point.
+That file uses the bash internal ``source`` to run several helper scripts, which
+are located in the ``common/`` subdirectory.
+
+
+Assumptions
+-----------
+
+The script makes the following assumptions:
+
+1. the script is being run on the Salt Master of a Ceph cluster deployed using
+   ceph-bootstrap
+2. the script is being run as root
+3. the Ceph admin keyring is installed in the usual way, so the root user can
```

Comment:
Does this mean that I have to manually install and configure `ceph client` on `Salt Master` to run QA tests?

Reply:
Not sure what you mean by `ceph client` . . . I have tested this using

https://github.com/SUSE/sesdev/pull/46

and it "just works", no manual steps required.

Reply:
I have a `sesdev` env that I've created using the `--ceph-bootstrap-repo` flag, and I got the following error when executing qa tests from `Salt Master`:

```
admin:~ # /root/ceph-bootstrap/qa/health-ok.sh
Running 'getopt --test'... PASS

WWWW: ceph_version_test
+ rpm -q ceph-common
package ceph-common is not installed
+ catch 1
+ echo

+ echo -n 'Overall result: '
Overall result: + '[' 1 = 0 ']'
+ echo 'NOT_OK (error )'
NOT_OK (error )
```

Reply:
@ricardoasmarques You have to use ceph-bootstrap to deploy a Ceph (SES7) cluster, first, before running `health-ok.sh`.

Maybe something has changed, but the last time I tried it, deploying SES7 caused `ceph-common` to be installed on the Salt Master. If this is no longer the case, this PR will need a modification.

Reply:
@smithfarm `ceph-common` is only installed in the minions with the `mon` role. Maybe you've tested in a single-node cluster where the salt-master is also a `mon`.

Reply:
Does `ceph-bootstrap` have an "admin" role? My expectation is I can do `ceph -s` from the admin node.

Reply:
To run QA tests we need both `ceph-bootstrap` and `ceph` commands.
Given that `ceph-bootstrap` is installed on `Salt Master` and `ceph-common` is installed on `mon`s, maybe we have to execute ceph commands via ssh, e.g.: `ssh node1 ceph health`.

Reply:
No. The assumption is that you will have an admin keyring and ceph-common installed in all `mon` minions.